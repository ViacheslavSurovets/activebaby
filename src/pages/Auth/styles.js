import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';
import { commonStyleTextThreeRem, flexCommon } from '@commonStyles';
import { commonStyleTextFourRem, commonFontColor, commonLinkStyle, commonStyleTextTwoRem } from '@commonStyles';
import { mobile, tablet } from '@media';


export const LoginContainer = styled.div`
  ${ flexCommon };
  flex-direction: column;
  min-height: 80vh;
  width: 90%;
  margin: 0 auto;
`;

export const LoginContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 40%;
  ${ tablet }{
    width:75%;
  };
  ${ mobile }{
    width: 90%;
  }
`;

export const LoginContainerButtonWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
`;

export const LoginContainerTitle = styled.div`
  ${ commonStyleTextFourRem };
  ${ commonFontColor };
  align-self: flex-start;
  margin-bottom: 6rem;
  ${ mobile }{
    ${ commonStyleTextThreeRem };
  }
`;

const styleLoginLink = css`
  ${ commonLinkStyle };
   color: ${ ( { theme } ) => theme.orangeSoft };
   &:hover {
    color: ${ ( { theme } ) => theme.blue }
  }
`;

export const RestorePasswordLink = styled ( Link )`
  ${ styleLoginLink };
  position: absolute;
  right: 0;
  transform: translateY(-110%);
  font-size: 1.8rem;
  font-style: italic;
  white-space: nowrap;
  ${ mobile }{
    bottom:0;
    transform: translateY(-60%);
  }
`;

export const LoginNewCustomerContainer = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  height: 5rem;
  margin-top: 1rem;
  ${ tablet },
  ${ mobile }{
    flex-wrap: wrap;
  }
`;

export const LoginNewCustomerText = styled.span`
  ${ commonStyleTextTwoRem };
  ${ commonFontColor };
  margin-right: 1rem;
  white-space: nowrap;
  ${ mobile }{
    white-space: inherit;
  }
`;

export const LoginNewCustomerLink = styled ( Link )`
  ${ styleLoginLink };
  ${ flexCommon };
  ${ commonStyleTextThreeRem };
`;
