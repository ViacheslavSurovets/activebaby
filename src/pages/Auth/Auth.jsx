import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { LoginForm, RegistrationForm } from './internal';
import { LoginContainer } from './styles';

import PropTypes from 'prop-types';

const Auth = ( props ) => {
  const { match } = props;

  return (
    <LoginContainer>
      <Switch>
        <Route exact path={ `${ match.path }` } component={ LoginForm } />
        <Route path={`${match.path}/registration`} component={RegistrationForm} />
      </Switch>
    </LoginContainer>
  );
};

export default withRouter(Auth);

Auth.propTypes = {
  match: PropTypes.object.isRequired
};
