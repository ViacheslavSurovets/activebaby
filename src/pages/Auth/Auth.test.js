import React from 'react';
import { shallow } from 'enzyme';
import Auth from './Auth';


describe ( 'Auth', () => {

  it ( 'Auth Snapshot', () => {
    expect ( shallow ( <Auth  /> ) ).toMatchSnapshot ();
  } );
} );
