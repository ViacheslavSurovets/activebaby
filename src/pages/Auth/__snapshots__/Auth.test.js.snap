// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Auth Snapshot 1`] = `ShallowWrapper {}`;
